sdk/python/kfp/compiler/compiler.py

```diff
@@ -562,13 +562,16 @@ def _compile(self, pipeline_func):
     workflow = self._create_pipeline_workflow(args_list_with_defaults, p)
     return workflow
 
-  def compile(self, pipeline_func, package_path):
+  def compile(self, pipeline_func, package_path, type_check=False):
```

Comment:
Could you add at least one test for each public interface?

Reply:
Added.